packages/web-components/src/components/va-pagination/va-pagination.tsx

```diff
@@ -338,71 +391,136 @@ export class VaPagination {
-      page < pages ? (
-        <Fragment>
-          <li class={arrowClasses}>
-            <a
-              aria-label={nextAriaLabel}
-              onClick={() =>
-                this.handlePageSelect(page + 1, 'nav-paginate-number')
-              }
-              onKeyDown={e => this.handleKeyDown(e, page + 1)}
-              class="usa-pagination__link usa-pagination__next-page"
-              href="javascript:void(0)"
-            >
-              <span class="usa-pagination__link-text">{i18next.t('next')}</span>
-              <div id="next-arrow-icon"></div>
-            </a>
-          </li>
-        </Fragment>
-      ) : null;
+
+    return renderedMiddlePages;
+  }
+
+  private renderLastPage(
+    ellipsisClasses: string,
+    itemClasses: string,
+  ): JSX.Element {
+    // Last page has already been rendered with renderMiddlePages
+    if (this.pageNumbers().includes(this.pages)) return null;
+
+    let lastEllipsis = null;
+    let lastPageButton = null;
+
+    if (this.pages > this.SHOW_ALL_PAGES) {
+      lastEllipsis = (
+        <li class={ellipsisClasses}>
+          <span class="usa-sr-only">Ellipsis indicating non-visible pages</span>
+          <span aria-hidden="true">…</span>
+        </li>
+      );
+    }
+
+    if (!this.unbounded && this.pages > this.SHOW_ALL_PAGES) {
+      lastPageButton = (
+        <li class={itemClasses}>
+          <a
+            onClick={() =>
+              this.handlePageSelect(this.pages, 'nav-paginate-number')
+            }
+            onKeyDown={e => this.handleKeyDown(e, this.pages)}
+            href="javascript:void(0)"
+            class="usa-pagination__button"
+            aria-label={`page ${this.pages}, last page`}
+          >
+            {this.pages}
+          </a>
+        </li>
+      );
+    }
+
+    return (
+      <Fragment>
+        {lastEllipsis}
+        {lastPageButton}
+      </Fragment>
+    );
+  }
+
+  private renderNextPageButton(
+    arrowClasses: string,
+    nextAriaLabel: string,
+  ): JSX.Element {
+    // Skip rendering if the viewport is too narrow to show Next button
+    if (this.isMobileViewport) return null;
+
+    // Return early if we're on the last page or current page was entered incorrectly
+    if (this.page >= this.pages) return null;
+
+    return (
+      <li class={arrowClasses}>
+        <a
+          aria-label={nextAriaLabel}
+          onClick={() =>
+            this.handlePageSelect(this.page + 1, 'nav-paginate-number')
+          }
+          onKeyDown={e => this.handleKeyDown(e, this.page + 1)}
+          class="usa-pagination__link usa-pagination__next-page"
+          href="javascript:void(0)"
+        >
+          <span class="usa-pagination__link-text">{i18next.t('next')}</span>
+          <div id="next-arrow-icon"></div>
+        </a>
+      </li>
+    );
+  }
+  /**
+   * =========================================
+   * End pagination render functions
+   * =========================================
```

Comment:
Nit: I'm personally not a fan of adding these large comment blocks as waypoints for code sections (here and on line 301). Function names should already be identifying what they're for and it seems like they do.

Reply:
Not a problem, I'll take them out in just a short.